### ISSUE TITLE
Instance / sample weights with boosted regression tree

### ISSUE THREAD

**Issue by fjanoos** [dmlc/xgboost#520]
Hello,

Does the boosted regression tree implementation (objective='reg:linear') in xgboost support sample weights ? I've browsed through the code and tried some experiments, but it does not seem to be the case. 

Please advise.

Thanks,
-firdaus

**Comment by tqchen** [comment]
Yes, checkout for example https://github.com/dmlc/xgboost/blob/master/demo/kaggle-higgs/higgs-numpy.py#L28

**Comment by fjanoos** [comment]
Thanks for the quick response. Starting with this example earlier, I replaced objective = 'binary:logitraw' with objective='reg:linear'. 

For this objective function, the results with and without weights seem to be the same.

Any ideas why that might be?
Thanks.

**Comment by tqchen** [comment]
Could simply because the weight do not contributed to the regression loss as much. The weight contributes to the loss function in all cases

**Comment by fjanoos** [comment]
Ah - you're right. I had to scale up the weights to see a difference. Thanks !
